repository_service_tuf/helpers/tuf.py

```diff
@@ -185,18 +180,23 @@ def _update_snapshot(targets_meta: List[Tuple[str, int]]) -> int:
             for signer in signers
         ]
 
-    root = Root(roles=roles)
-    for arg in add_key_args:
-        root.add_key(arg[0], arg[1])
-
     # Add signature wrapper, bump expiration, and sign and persist
-    for role in [targets, snapshot, timestamp, root]:
-        # TODO: Create an issue for this mypy error if it's not resolved by the
-        # PR reviews
-        metadata = Metadata(role)  # type: ignore
-        _bump_expiry(metadata, role.type)
-        _sign(metadata, role.type)
-        _add_payload(metadata, role.type)
+    for role in [Targets, Snapshot, Timestamp, Root]:
+
+        # Bootstrap default top-level metadata to be updated below if necessary
+        if role is Root:
+            metadata = Metadata(role(roles=roles))
```

Comment:
I think you can replace `role` with `Root` to make it clearer:
```suggestion
            metadata = Metadata(Root(roles=roles))
```